app/services/katello/pulp3/repository/yum.rb

```diff
@@ -104,14 +100,32 @@ def copy_units(source_repository, content_unit_hrefs, dependency_solving, dest_b
                 }
               end
             end
+            data_package_env = copy_package_env(source_repository, dest_base_version)
             tasks << api.copy_api.copy_content(data)
+            tasks << api.copy_api.copy_content(data_package_env) if (smart_proxy.pulp3_copy_rpm_package_env? && data_package_env)
           else
             tasks << remove_all_content
           end
-
           tasks
         end
 
+        def copy_package_env(source_repository, dest_base_version)
+          options = { :repository_version => source_repository.version_href }
+          package_env_hrefs = packageenvironments(options).results.map(&:pulp_href)
```

Comment:
could there be more than a page_size of results?  Probably want to use this:  https://github.com/Katello/katello/blob/master/app/services/katello/pulp3/api/core.rb#L145

Reply:
oh and you might wanna sort href list that avoid testing issues :)